tests/kdb/testkdb_highlevel.cpp

```diff
@@ -12,6 +12,7 @@
 
 #include <gtest/gtest-elektra.h>
 #include <kdbhelper.h>
+#include <kdbprivate.h>
```

Comment:
We also should have ABI tests, i.e. tests that only use the public interface.

Reply:
The tests themselves do only access the public interface. Some use the low-level API too to check the results, but `kdbprivate.h` is only used for the fatal error handler that outputs error messages. In particular we only use the `elektraError`* and `elektraKDBError*` functions that were part of the public interface before this PR.

Reply:
For ABI tests it would be needed that no methods of kdbprivate.h is used at all. Otherwise the ABI tests would also fail on changes within kdbprivate.h.